packages/astro/src/vite-plugin-astro-server/index.ts

```diff
@@ -248,26 +248,81 @@ async function handleRequest(
 		clientAddress: buildingToSSR ? req.socket.remoteAddress : undefined,
 	});
 
+	async function matchRoute() {
+		const matches = matchAllRoutes(pathname, manifest);
+
+		if (config.output === 'server' && matches.length > 1) {
+			throw new Error(`Found multiple matching routes for "${pathname}"! When using \`output: 'server'\`, only one route in \`src/pages\` can match a given URL. Found:
+
+${
+	matches.map(({ component }) => `- ${component}`).join('\n')
+}
+`);
+		}
+
+		for await (const maybeRoute of matches) {
+			const filePath = new URL(`./${maybeRoute.component}`, config.root);
+			const preloadedComponent = await preload({ astroConfig: config, filePath, viteServer });
+			const [, mod] = preloadedComponent;
+			// attempt to get static paths
```

Comment:
does this result in getStaticPaths being called more than once?

Reply:
It's still only called once.  The `getStaticPaths()` result is always cached even if the params don't match

Just noticed we have a test to make sure it's only called once but that test only covers `build`.  I just pushed up a test that checks that for `dev` as well :+1: